### ISSUE TITLE
Remove namecoin from bnsd (and bcpd?) apps

### ISSUE THREAD

**Issue by ethanfrey** [iov-one/weave#198]
We are using NFTs to handle username registration. We can now remove the temporary namecoin solution.

We don't need to remove the extension, it is at least an interesting example of extending cash, but we should not use the namecoin extension in these apps, but rather the simpler cash extension.

**Comment by ruseinov** [comment]
makes sense, will do it as part of my batch PR as I’m changing messages quite a bit there if nobody minds.

**Comment by ruseinov** [comment]
although, maybe easier to do in a small pr

**Comment by ethanfrey** [comment]
It should be a separate PR
No rush on it, just shouldn't be forgotten

**Comment by husio** [comment]
@ethanfrey can you please explain a bit more what is expected to be done in this issue? My understanding of it is that `x/namecoin` is to be removed from both `bnsd` and `bcpd` applications. We want to preserve the functionality though and build it directly on `x/cash`. Why would we do this, if the functionality is implemented in `namecoin`? Should something be done different? Maybe we need only some of the `namecoin` package functionality, but then we could split it.

**Comment by ethanfrey** [comment]
Namecoin extends cash and adds a very early prototype of name registration, which allowed up to build apps using it.

This functionality (SetName) is no longer needed as we have Nfts, and exposing it can lead to confusion with two different naming schemes.

Removing namecoin and replacing it with cash should only remove the support for `SetName` and associated queries. We don't need to migrate data, this is just about reducing the exposed surface area of the blockchain API.

btw, namecoin more or less "inherits" cash and extends it. You can look at it for some ideas how I tried to make extensible modules. When we need such functionality again, I would happy to have new design discussion on better ways to do so

**Comment by husio** [comment]
If I understand correctly, this is about removing `x/namecoin` because functionality it offers is no longer needed -- it's either replaced with what `x/nft` provides or just not wanted anymore. Is that correct?

If this is only about removing `x/namecoin` and not providing any replacement, than I think I'm good to go!

**Comment by ruseinov** [comment]
@husio please make sure you based this work off of batch branch until it's merged, as otherwise we're looking at broken code and conflicts.

**Comment by ruseinov** [comment]
@husio Disregard, it's merged!
